interfaces/cython/cantera/test/test_reactor.py

```diff
@@ -600,6 +660,52 @@ def test_pressure_controller(self):
             dP = self.r1.thermo.P - outlet_reservoir.thermo.P
             self.assertNear(mdot(t) + 1e-5 * dP, pc.mdot(t))
 
+    def test_pressure_controller2(self):
+        self.make_reactors(n_reactors=1)
+        g = ct.Solution('h2o2.xml')
+        g.TPX = 500, 2*101325, 'H2:1.0'
+        inlet_reservoir = ct.Reservoir(g)
+        g.TP = 300, 101325
+        outlet_reservoir = ct.Reservoir(g)
+
+        mfc = ct.MassFlowController(inlet_reservoir, self.r1)
+        mdot = lambda t: np.exp(-100*(t-0.5)**2)
+        mfc.mass_flow_coeff = 1.
+        mfc.set_time_function(mdot)
+
+        pc = ct.PressureController(self.r1, outlet_reservoir)
+        pc.set_master(mfc)
+        pfunc = lambda dp: 1.e-5 * abs(dp)**.5
+        pc.set_pressure_function(pfunc)
+        self.assertEqual(pc.pressure_coeff, 1.)
+
+        t = 0
+        while t < 1.0:
+            t = self.net.step()
+            self.assertNear(mdot(t), mfc.mdot(t))
+            dP = self.r1.thermo.P - outlet_reservoir.thermo.P
+            self.assertNear(mdot(t) + pfunc(dP), pc.mdot(t))
+
+    def test_pressure_controller_deprecations(self):
+        # Make sure Python deprecation warnings actually get displayed
+
+        self.make_reactors()
+        res = ct.Reservoir(self.gas1)
+        mfc = ct.MassFlowController(res, self.r1, mdot=0.6)
+
+        p = ct.PressureController(self.r1, self.r2, master=mfc, K=0.5)
+
+        with warnings.catch_warnings(record=True) as w:
```

Comment:
I didn't know you could do this, so thanks for teaching me something. :sunglasses:

Reply:
NP. [stackoverflow](https://stackoverflow.com/questions/3892218/how-to-test-with-pythons-unittest-that-a-warning-has-been-thrown) taught me ;)